packages/shared/src/components/sidebar/Sidebar.tsx

```diff
@@ -156,10 +163,35 @@ export default function Sidebar({
     optOutWeeklyGoal,
   } = useContext(SettingsContext);
   const [showSettings, setShowSettings] = useState(false);
+  const [showIntroModal, setShowIntroModal] = useState(false);
+
   const shouldShowDnD = !!process.env.TARGET_BROWSER;
   const { flags } = useContext(FeaturesContext);
   const popularFeedCopy = getFeatureValue(Features.PopularFeedCopy, flags);
   const feedFilterModal = getFeatureValue(Features.FeedFilterModal, flags);
+  const feedFilterModalOnboarding = getFeatureValue(
+    Features.FeedFilterModalOnboarding,
+    flags,
+  );
+  const FIRST_TIME_SESSION = 'firstTimeSession';
+
+  const [isFirstSession, setIsFirstSession] = usePersistentContext(
+    FIRST_TIME_SESSION,
+    true,
+  );
+  useEffect(() => {
+    if (isFirstSession) {
+      setIsFirstSession(false);
+      if (feedFilterModalOnboarding === 'control') {
+        setTimeout(() => {
+          openFeedFilters();
+        }, 300);
+      } else {
+        openFeedFilters();
+        setShowIntroModal(true);
+      }
+    }
+  }, [isFirstSession]);
```

Comment:
We don't want to listen to this change, it can simply fire once, so you can use a empty array `[]`